s3/pkg/datastore/yig/crypto/client.go

```diff
@@ -0,0 +1,26 @@
+package crypto
+
+import (
+	"github.com/opensds/multi-cloud/s3/pkg/helper"
+)
+
+func NewKMS() KMS {
+	switch helper.CONFIG.KMS.Type {
+	case "vault":
```

Comment:
kindly reach out when convenient to you, we can discuss this. In my analysis, we would NOT need to create a KMS instance ourselves...the KMS is hosted by the cloud provider like GCP, Azure or AWS, we just use it

Reply:
ok, thanks fine :)